compiler-core/src/build/package_compiler.rs

```diff
@@ -37,6 +37,7 @@ pub struct PackageCompiler<'a, IO> {
     pub copy_native_files: bool,
     pub compile_beam_bytecode: bool,
     pub silence_subprocess_stdout: bool,
+    pub write_build_journal: bool,
```

Comment:
Rather than have a boolean and also pass in the hashset as an argument we could store it here optionally. If present then we are to write, if not then we do not.

```suggestion
    /// If present any package artefacts written by the compiler will be recorded
    /// in this data structure.
    pub build_journal: HashSet<&'a mut PathBuf>,
```

Reply:
A- Do you mean this right?
``` 
pub build_journal: &'a mut HashSet<PathBuf>,
```

Reply:
Yes, sorry